src/box/mp_error.h

```diff
@@ -115,6 +115,19 @@ mp_snprint_error(char *buf, int size, const char **data, int depth);
 int
 mp_fprint_error(FILE *file, const char **data, int depth);
 
+/**
+ * Check that the buffer contains a valid packed error representation.
+ * @param data MessagePack encoded error, without MP_EXT header.
+ * @param len  Length of @a data.
+ * @retval 1   Error. The buffer doesn't contain a parseable error.
+ * @retval 0   Ok.
+ */
+static inline int
+mp_check_error(const char *data, uint32_t len)
+{
+	return error_unpack(&data, len) == NULL;
```

Comment:
1. AFAICS `error_unpack` will crash on [this](https://github.com/tarantool/tarantool/blob/f1d750f036956873c79bf5de51f77b38da25ae0f/src/box/mp_error.h#L70) assertion on error, because `error_unpack_unsafe` doesn't advance the data pointer on error.
2. On success the error created by `error_unpack` will leak.

Reply:
Thanks! I missed both of these. Fixed now.